src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3TlsTests.cs

```diff
@@ -61,6 +63,179 @@ public async Task ServerCertificateSelector_Invoked()
+                    });
+                });
+            });
+
+            using var host = builder.Build();
+            using var client = Http3Helpers.CreateClient(includeClientCert: true);
+
+            await host.StartAsync().DefaultTimeout();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"https://127.0.0.1:{host.GetPort()}/");
+            request.Version = HttpVersion.Version30;
+            request.VersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.SendAsync(request, CancellationToken.None).DefaultTimeout());
+            // This poor error is likely a symptom of https://github.com/dotnet/runtime/issues/57246
+            // QuicListener returns the connection before (or in spite of) the cert validation failing.
+            // There's a race where the stream could be accepted before the connection is aborted.
+            var qex = Assert.IsType<QuicOperationAbortedException>(ex.InnerException);
+            Assert.Equal("Operation aborted.", qex.Message);
+
+            await host.StopAsync().DefaultTimeout();
+        }
+
+        [ConditionalFact]
+        [MsQuicSupported]
+        public async Task ClientCertificate_Allow_NotAvailable_Optional()
+        {
+            var builder = CreateHostBuilder(async context =>
+            {
+                var hasCert = context.Connection.ClientCertificate != null;
+                await context.Response.WriteAsync(hasCert.ToString());
+            }, configureKestrel: kestrelOptions =>
+            {
+                kestrelOptions.ListenAnyIP(0, listenOptions =>
+                {
+                    listenOptions.Protocols = HttpProtocols.Http3;
+                    listenOptions.UseHttps(httpsOptions =>
+                    {
+                        httpsOptions.ServerCertificate = TestResources.GetTestCertificate();
+                        httpsOptions.ClientCertificateMode = ClientCertificateMode.AllowCertificate;
+                        httpsOptions.AllowAnyClientCertificate();
+                    });
+                });
+            });
+
+            using var host = builder.Build();
+            using var client = Http3Helpers.CreateClient(includeClientCert: false);
+
+            await host.StartAsync().DefaultTimeout();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"https://127.0.0.1:{host.GetPort()}/");
+            request.Version = HttpVersion.Version30;
+            request.VersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+            // https://github.com/dotnet/runtime/issues/57308, optional client certs aren't supported.
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(() => client.SendAsync(request, CancellationToken.None).DefaultTimeout());
+            Assert.StartsWith("Connection has been shutdown by transport. Error Code: 0x80410100", ex.Message);
+            /*
```

Comment:
Meant to still be here?

Reply:
Someday this will be fixed, but in the meantime I'll remove that code.